### PR TITLE
Use ginkgo's context in e2e

### PR DESCRIPTION
**Pull request by nunnatsa**
## What this PR does / why we need it
Use ginkgo contexts instead and let ginkgo handle them, instead of creating our on contexts.

This is prove of concept, so it's done for several tests, but not for all of them.

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by nunnatsa** [comment]
/hold

looks like a ginkgo issue: https://github.com/onsi/ginkgo/issues/1415

**Review by tiraboschi — APPROVED** [review]

**Comment by nunnatsa** [comment]
/unhold

@tiraboschi could you PTAL?

File: envs.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
...
...:["registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-hostpath-provisioner-rhel9@sha256:c21f65ff35d256f2d9ab85471a27f4c10b43216924af2b0cfd29c04372e37bc2","registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-hostpath-provisioner-rhel9:v4.16.0-150"],"digests":{"application/vnd.docker.distribution.manifest.list.v2+json":"sha256:c21f65ff35d256f2d9ab85471a27f4c10b43216924af2b0cfd29c04372e37bc2"}},"media_types":["application/vnd.docker.distribution.manifest.list.v2+json","application/vnd.docker.distribution.manifest.v1+json","application/vnd.docker.distribution.manifest.v2+json"]}},"custom_user_metadata":{"cpaas_pipeline_id":"184cb8ce-67e3-3de0-b594-40bb70ff93f7"}},"id":3038090,"name":"hostpath-provisioner-rhel9-container","nvr":"hostpath-provisioner-rhel9-container-v4.16.0-150","owner_id":6716,"owner_name":"exd-cpaas-bot-prod","package_id":81529,"package_name":"hostpath-provisioner-rhel9-container","promoter_id":null,"promoter_name":null,"promotion_time":null,"promotion_ts":null,"release":"150","source":"https://pkgs.devel.redhat.com/git/containers/hostpath-provisioner#a586be55fdea052dc40142bf0154e1d04eb37f72","start_time":"2024-05-02
```
Comment (tiraboschi) [inline envs.txt:5]: Do we need this?

File: np.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+          Machine second-vntkg-d897x: DVNotReady: DataVolume second-b1dd31ce-jjn4p-rhcos import is not
+          running: Unable to process data: Unable to convert source data to target format: qemu-img: 
+          Could not open 'nbd+unix:///?socket=/tmp/nbdkit.sock': Requested export not available , 
+          qemu-img execution failed: exit status 1 Log line from nbdkit: nbdkit: curl[1]: error: problem 
+          doing HEAD request to fetch size of URL [http://localhost:8100/rhcos-416.94.202404301731-0-kubevirt.x86_64.qcow2]: 
+          Couldn't resolve proxy name: Could not resolve proxy: fakeurl
+        observedGeneration: 1
+        reason: DVNotReady
+        status: "False"
+        type: AllMachinesReady
```
Comment (tiraboschi) [inline np.yaml:18]: same

**Pull request merged**